packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/mtom-parser.ts

```diff
@@ -0,0 +1,149 @@
+import { XMLParser } from "fast-xml-parser";
+import { DocumentResponse } from "./dr-response";
+import { XML_APP_MIME_TYPE, XML_TXT_MIME_TYPE } from "../../../../../../util/mime";
+import { stripCidPrefix, stripTags } from "../../../utils";
+
+const quoteRegex = /"/g;
+const carriageReturnLineFeed = "\r\n\r\n";
+
+type MtomContentType = {
+  boundary: string;
+  start: string;
+  type?: string | undefined;
+  startInfo?: string | undefined;
+};
+
+function parseMtomContentType(contentType: string): MtomContentType {
+  const contentTypeParams = contentType.split(";").reduce<Record<string, string>>((acc, param) => {
+    const index = param.indexOf("=");
+    if (index !== -1) {
+      const key = param.substring(0, index).trim().toLowerCase();
+      const value = param
+        .substring(index + 1)
+        .trim()
+        .replace(quoteRegex, "");
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+
+  if (!contentTypeParams.boundary) {
+    throw new Error("No boundary parameter found in content type.");
+  }
+  if (!contentTypeParams.start) {
+    throw new Error("No start parameter found in content type.");
+  }
+  return {
+    boundary: contentTypeParams.boundary,
+    type: contentTypeParams.type,
+    start: contentTypeParams.start,
+    startInfo: contentTypeParams["start-info"],
+  };
+}
+
+type MtomHeaders = {
+  ContentID: string;
+  ContentTransferEncoding?: string | undefined;
+  ContentType: string;
+};
+
+function parseMtomHeaders(headerPart: string): MtomHeaders {
+  const headers = headerPart.split("\n").reduce<Record<string, string>>((acc, headerLine) => {
```

Comment:
Can't we use JS' `Headers` for to parse these? https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers

Reply:
I don't see anything in what you linked which will help with this case. I am going to look into all this stuff for the follow on PR since we need to rework things anyway to get non-xml data types working: https://github.com/metriport/metriport-internal/issues/1784

Reply:
Yeah, I thought the `Headers` constructor accepted a string version of it, but it doesn't.